setfit blog draft v0 (#525)
huggingface/blog: setfit.md

---
title: "SetFit: Efficient Few-Shot Learning Without Prompts"
thumbnail: /blog/assets/103_setfit/intel_hf_logo.png
---
<div class="blog-metadata">
    <small>Published September 24, 2022.</small>
   
<h1>SetFit: Efficient Few-Shot Learning Without Prompts</h1>

</div>

<div class="author-card">
    <a href="/unsojo">
        <div class="bfc">
            <code>unsojo</code>
            <span class="fullname">Unso Jo</span>
        </div>
  </a>
</div>

# SetFit: Efficient Few-Shot Learning Without Prompts

> ### Hugging Face, Intel labs, and the UKP team introduce a new prompt-free few-shot learning regime called Setfit with accompanying [code](https://github.com/SetFit/setfit), longer [paper](), and datasets on [HF hub](https://huggingface.co/SetFit)

Reviewer: I wouldn't use this text as a subheader. Instead, I suggest using a hook to excite the reader, e.g. something like: 

Few-shot learning with pretrained language models has emerged as a promising solution to every data scientist's nightmare: dealing with data that has few to no labels 😱. 

Together with our research partners at [Intel Labs](https://www.intel.com/content/www/us/en/research/overview.html) and the [UKP Lab](https://www.informatik.tu-darmstadt.de/ukp/ukp_home/index.en.jsp), we are excited to introduce SetFit: an efficient framework for few-shot fine-tuning of Sentence Transformers. 

Compared to other few-shot learning methods, SetFit has several unique features:

* 📈 **High accuracy with little labeled data:** SetFit achieves comparable (or better) results than current state-of-the-art methods for text classification. For example, with only 8 labelled examples per class on the CR sentiment dataset, SetFit is competitive with fine-tuning RoBERTa-large on the full training set of 3k examples.
* 🗣 **No prompts or verbalisers:** Current techniques for few-shot fine-tuning require handcrafted prompts or verbalisers to convert examples into a format that's suitable for the underlying language model. SetFit dispenses with prompts altogether by generating rich embeddings directly from text examples.
* 🏎 **Fast to train:** SetFit doesn't require large-scale models like T0 or GPT-3 to achieve high accuracy. As a result, it is typically an order of magnitude (or more) faster to train and run inference with.

For more details, you can check out our paper (ADD LINK) and code (ADD LINK). In this blog post, we'll explain how SetFit works and how to train your very own models. Let's dive in!

## How does it work?

[High level overview, with image of process from paper]

[Show Fig. 1 from the paper as comparison to standard fine tuning]

## Benchmarking SetFit

[Idea: pick 3 datasets from Table 2 and create box plots from the `final_results` to show the spread of values visually. If time doesn't allow, just show the table]

### Fast training and inference

[Idea: convert table 5 to 2 plots, with score on y-axis, FLOPs on x-axis, and model size as radius of circle]

## Training your own model

[Show quickstart]

---
title: "SetFit: Efficient Few-Shot Learning Without Prompts"
thumbnail: /blog/assets/103_setfit/intel_hf_logo.png
---
<div class="blog-metadata">
    <small>Published September 24, 2022.</small>
   
<h1>SetFit: Efficient Few-Shot Learning Without Prompts</h1>

</div>
<div class="author-card">
    <a href="/Unso">
        <img class="avatar avatar-user" src="https://scholar.googleusercontent.com/citations?view_op=medium_photo&user=-I2EZeEAAAAJ&citpid=8" title="Gravatar">
        <div class="bfc">
            <code>Unso</code>
            <span class="fullname">Unso Eun Seo Jo</span>
        </div>
    </a>
</div>

Few-shot learning with pretrained language models has emerged as a promising solution to every data scientist's nightmare: dealing with data that has few to no labels 😱.

> Together with our research partners at [Intel Labs](https://www.intel.com/content/www/us/en/research/overview.html) and the [UKP Lab](https://www.informatik.tu-darmstadt.de/ukp/ukp_home/index.en.jsp), we are excited to introduce SetFit: an efficient framework for few-shot fine-tuning of Sentence Transformers.

Reviewer: I realised that our first comparison point to few-shot methods ("High accuracy with little labeled data") isn't quite accurate since they also do well. Instead I suggest we remove this point and add the following sentence:

```suggestion
Together with our research partners at [Intel Labs](https://www.intel.com/content/www/us/en/research/overview.html) and the [UKP Lab](https://www.informatik.tu-darmstadt.de/ukp/ukp_home/index.en.jsp), we are excited to introduce SetFit: an efficient framework for few-shot fine-tuning of [Sentence Transformers](https://sbert.net/). SetFit achieves high accuracy with little labeled data - for example, with only 8 labeled examples per class on the Customer Reviews sentiment dataset, SetFit is competitive with fine-tuning RoBERTa Large on the full training set of 3k examples 🤯!

<p align="center">
    <img src="assets/103_setfit/setfit_curves.png" width=400>
</p>
<p align="center">
    SetFit and fine-tuning performance on the Customer Reviews sentiment dataset.
</p>
```

Compared to other few-shot learning methods, SetFit has several unique features:

> <p>📈 <strong>High accuracy with little labeled data</strong>: SetFit achieves comparable (or better) results than current state-of-the-art methods for text classification. For example, with only 8 labelled examples per class on the CR sentiment dataset, SetFit is competitive with fine-tuning RoBERTa-large on the full training set of 3k examples. </p>

Reviewer: Delete this point as suggested above:

```suggestion
```

> <p>🗣 <strong>No prompts or verbalisers</strong>: Current techniques for few-shot fine-tuning require handcrafted prompts or verbalisers to convert examples into a format that's suitable for the underlying language model. SetFit dispenses with prompts altogether by generating rich embeddings directly from text examples. </p>

Reviewer: ```suggestion
<p>🗣 <strong>No prompts or verbalisers</strong>: Current techniques for few-shot fine-tuning require handcrafted prompts or verbalisers to convert examples into a format that's suitable for the underlying language model. SetFit dispenses with prompts altogether by generating rich embeddings directly from a small number of labeled text examples. </p>
```

<p>🏎 <strong>Fast to train</strong>: SetFit doesn't require large-scale models like T0 or GPT-3 to achieve high accuracy. As a result, it is typically an order of magnitude (or more) faster to train and run inference with. </p>

> <p>🌎 <strong>Multilingual support</strong>: SetFit can be used with any Sentence Transformer on the Hub, which means you can classify text in multiple languages by simply switching the backbone to a multilingual checkpoint. </p>

Reviewer: ```suggestion
<p>🌎 <strong>Multilingual support</strong>: SetFit can be used with any [Sentence Transformer](https://huggingface.co/models?library=sentence-transformers&sort=downloads) on the Hub, which means you can classify text in multiple languages by simply fine-tuning a multilingual checkpoint. </p>
```

> For more details, you can check out our [paper](https://arxiv.org/abs/2209.11055), [data](https://huggingface.co/SetFit) and [code](https://github.com/huggingface/setfit). In this blog post, we'll explain how SetFit works and how to train your very own models. Let's dive in!

Reviewer: turbo nit:
```suggestion
For more details, you can check out our [paper](https://arxiv.org/abs/2209.11055), [data](https://huggingface.co/SetFit), and [code](https://github.com/huggingface/setfit). In this blog post, we'll explain how SetFit works and how to train your very own models. Let's dive in!
```

> For more details, you can check out our [paper](https://arxiv.org/abs/2209.11055), [data](https://huggingface.co/SetFit) and [code](https://github.com/huggingface/setfit). In this blog post, we'll explain how SetFit works and how to train your very own models. Let's dive in!

Reviewer: ```suggestion
For more details, check out our [paper](https://arxiv.org/abs/2209.11055), [data](https://huggingface.co/SetFit) and [code](https://github.com/huggingface/setfit). In this blog post, we'll explain how SetFit works and how to train your very own models. Let's dive in!
```

> <p align="center">

Reviewer: I suggest moving this image up, so let's delete this snippet here

    <img src="assets/103_setfit/setfit_curves.png" width=400>
</p>
<p align="center">
    Fig.1: SetFit and Fine-tuning performance on SentEval test set, a sentiment classification task
</p>

## How does it work?

> SetFit is designed with efficiency and simplicity in mind. SetFit first finetunes a Sentence Transformer model then trains a classifier head on the embeddings generated from the finetuned Sentence Transformer. 

Reviewer: ```suggestion
SetFit is designed with efficiency and simplicity in mind. SetFit first fine-tunes a Sentence Transformer model on a small number of labeled examples (typically 8 or 16 per class). This is followed by training a classifier head on the embeddings generated from the fine-tuned Sentence Transformer. 
```

<p align="center">
    <img src="assets/103_setfit/setfit_diagram_process.png" width=700>
</p>
<p align="center">

>     Fig.2: SetFit's 2-Part Process

Reviewer: ```suggestion
    SetFit's two-stage training process
```

</p>

> SetFit takes advantage of sentence transformers’ ability to generate dense embeddings based on paired sentences. In the data input stage, it maximizes the limited labeled input data by contrastive training, where positive and negative pairs are created by in-class and out-class selection. The Sentence Transformer model then trains on these pairs (or triplets) and generates dense vectors per example. This is SetFit’s fine-tuning step. In the second step, the classification head, such as a logistic regression model, trains on the encoded embeddings with their respective class labels. At inference time, the unseen example passes through the fine-tuned Sentence Transformer, generating an embedding that when fed to the classification head outputs a class label prediction.

Reviewer: ```suggestion
SetFit takes advantage of Sentence Transformers’ ability to generate dense embeddings based on paired sentences. In the data input stage, it maximizes the limited labeled input data by contrastive training, where positive and negative pairs are created by in-class and out-class selection. The Sentence Transformer model then trains on these pairs (or triplets) and generates dense vectors per example. This is SetFit’s fine-tuning step. In the second step, the classification head, such as a logistic regression model, trains on the encoded embeddings with their respective class labels. At inference time, the unseen example passes through the fine-tuned Sentence Transformer, generating an embedding that when fed to the classification head outputs a class label prediction.
```

And just by switching out the base Sentence Transformer model to a multilingual one, SetFit can function seamlessly in multilingual contexts. In our experiments, SetFit’s performance shows promising results on classification in German, Japanese, Mandarin, French and Spanish, in both in-language and cross linguistic settings.

## Benchmarking SetFit

| Rank | Method | Accuracy | Model Size | 
| :------: | ------ | :------: | :------: | 
| 2 | T-Few | 75.8 | 11B | 
| 4 | Human Baseline | 73.5 | N/A | 
| 6 | SetFit (Roberta Large) | 71.3 | 355M |
| 9 | PET | 69.6 | 235M |
| 11 | SetFit (MP-Net) | 66.9 | 110M |
| 12 | GPT-3 | 62.7 | 175 B |

> <h4>Table 1: RAFT performance leaderboard as of September 2022</h4>

Reviewer: ```suggestion
<h4>Prominent methods on the RAFT leaderboard (as of September 2022)</h4>
```

> Although based on much smaller models than existing few-shot methods, SetFit performs on par or better than state of the art few-shot regimes on a variety of benchmarks. On [RAFT](https://huggingface.co/spaces/ought/raft-leaderboard), a few-shot benchmark dataset as of September 2022, SetFit Roberta (using the Roberta-Large Sentence Transformer base model) with 355 million parameters outforms PET and GPT-3 and places just under average human performance and the 11 billion parameter T-few, a model 30 times the size of SetFit Roberta (Table 1). SetFit also outperforms the human baseline on 7 of the 11 RAFT tasks.

Reviewer: ```suggestion
Although based on much smaller models than existing few-shot methods, SetFit performs on par or better than state of the art few-shot regimes on a variety of benchmarks. On [RAFT](https://huggingface.co/spaces/ought/raft-leaderboard), a few-shot classification benchmark, SetFit Roberta (using the [`all-roberta-large-v1`](https://huggingface.co/sentence-transformers/all-roberta-large-v1) base model) with 355 million parameters outforms PET and GPT-3 and places just under average human performance and the 11 billion parameter T-few, a model 30 times the size of SetFit Roberta (Table 1). SetFit also outperforms the human baseline on 7 of the 11 RAFT tasks.
```

Reviewer: Should the `Setfit_RAFT.png` image appear below this paragraph?

<p align="center">
    <img src="assets/103_setfit/three-tasks.png" width=700>
</p>
<p align="center">

>     Fig.3: Comparing Setfit performance against other methods on 3 tasks

Reviewer: ```suggestion
    Comparing Setfit performance against other methods on 3 classification datasets.
```

</p>

> On other datasets, SetFit shows robustness across a variety of tasks. It outperforms PERFECT, T-Few 3B, ADAPET and fine-tuned vanilla transformers, on many tasks on sentiment, emotion, counterfactual, and unwanted language classification tasks at very few (n=8) and few (n=64) - shot learning scenarios. Figure 3 shows a comparison on three chosen tasks: Emotion, SentEval-CR, and Amazon Counterfactual. 

Reviewer: ```suggestion
On other datasets, SetFit shows robustness across a variety of tasks. It outperforms PERFECT, T-Few 3B, ADAPET and fine-tuned vanilla transformers, on many tasks on sentiment, emotion, counterfactual, and unwanted language classification tasks at very few (n=8) and few (n=64) - shot learning scenarios. Figure 3 shows a comparison on three chosen tasks: Emotion, CR, and Amazon Counterfactual. 
```

> On other datasets, SetFit shows robustness across a variety of tasks. It outperforms PERFECT, T-Few 3B, ADAPET and fine-tuned vanilla transformers, on many tasks on sentiment, emotion, counterfactual, and unwanted language classification tasks at very few (n=8) and few (n=64) - shot learning scenarios. Figure 3 shows a comparison on three chosen tasks: Emotion, SentEval-CR, and Amazon Counterfactual. 

Reviewer: ```suggestion
On other datasets, SetFit shows robustness across a variety of tasks. It outperforms standard fine-tuning, PERFECT, and ADAPET with 8 and 64 labeled examples per class, and is comparable to T-Few (the current SOTA). Figure 3 shows a comparison on three chosen tasks: Emotion, SentEval-CR, and Amazon Counterfactual. 
```

## Fast training and inference

<p align="center">
    <img src="assets/103_setfit/bars.png" width=400>
</p>
<p align="center">

>     Fig.4: Comparing T-Few 3B and SetFit (MP-Net) train cost on AWS for N=8 and average accuracy across test tasks

Reviewer: ```suggestion
    Comparing training cost and average performance for T-Few 3B and SetFit (MPNet), with 8 labeled examples per class.
```

</p>

Since SetFit achieves high accuracy with relatively small models, this makes it blazing fast to train on a single GPU like the ones found on Google Colab (in fact you can train SetFit on CPU in just a few minutes!). For instance, training SetFit on an NVIDIA V100 with 8 labeled examples takes just 30 seconds, at a cost of $0.025. By comparison, training T-Few 3B requires an NVIDIA A100 and takes 11 minutes, at a cost of around $0.7 for the same experiment - a factor of 28x more. As shown in Figure 4, this speed-up comes with comparable model performance. Similar gains are also achieved for inference - as we show in the paper, distilling the SetFit model can bring speed-ups of 123x 🤯.

## Training your own model

> Using SetFit is as simple as just a few lines of code. There is need for hyperparameter searching or prompt-engineering.

Reviewer: ```suggestion
To make SetFit accessible to the community, we've created a small `setfit` [library](https://github.com/huggingface/setfit) that allows you to train your own models with just a few lines of code. 
```

s

> To start using SetFit, first install it using pip. This will also install dependencies such as datasets and sentence-transformers. 

Reviewer: ```suggestion
The first thing to do is install it by running the following command:
```

```sh
pip install setfit
```

> We first import relevant functions. In particular we import SetFitModel and SetFitTrainer, two functions that have streamlined the SetFit procedure for us.

Reviewer: ```suggestion
Next, we import `SetFitModel` and `SetFitTrainer`, two core classes that streamline the SetFit training process:
```

```sh
from datasets import load_dataset
from sentence_transformers.losses import CosineSimilarityLoss

> from setfit.modeling import SetFitModel

Reviewer: I tweaked the lib so this is now possible (and simpler):
```suggestion
from setfit import SetFitModel, SetFitTrainer
```

> from setfit.trainer import SetFitTrainer

Reviewer: ```suggestion
```

```

> We download our dataset from the HF hub. In this case we start with the SentEval-CR task, the performance of which is shown in our Fig.1. 

Reviewer: ```suggestion
Now, let's download a text classification dataset from the Hugging Face Hub. We'll use the `SentEval-CR` dataset, which is a dataset of customer reviews: 
```

> ```sh

Reviewer: ```suggestion
```python
```

> # Load a dataset

Reviewer: ```suggestion
```

dataset = load_dataset("SetFit/SentEval-CR")
```

> Let's now select our N, the number of examples per class. In this case we make this 8. N will vary depending on your dataset constraints. 

Reviewer: ```suggestion
To simulate a real-world scenario with just a few labeled examples, we'll sample 8 examples per class from the training set: 
```

> ```sh

Reviewer: ```suggestion
```python
```

# Select N examples per class (8 in this case)
train_ds = dataset["train"].shuffle(seed=42).select(range(8 * 2))
test_ds = dataset["test"]
```

> We will load a pretrained Sentence Transformer model from the HF hub then create a SetFitTrainer. 

Reviewer: ```suggestion
Now that we have a dataset, the next step is to load a pretrained Sentence Transformer model from the Hub and instantiate a `SetFitTrainer`. Here we use the  `paraphrase-mpnet-base-v2` model, which we found to give great results across many datasets:
```

> ```sh

Reviewer: ```suggestion
```python
```

# Load SetFit model from Hub
model = SetFitModel.from_pretrained("sentence-transformers/paraphrase-mpnet-base-v2")

# Create trainer
trainer = SetFitTrainer(
    model=model,
    train_dataset=train_ds,
    eval_dataset=test_ds,
    loss_class=CosineSimilarityLoss,
    batch_size=16,
    num_epochs=20,
)
```

> Training this is easy! We can then evaluate with train.evaluate().

Reviewer: The last step is to train and evaluate the model:

> ```sh

Reviewer: ```suggestion
```python
```

# Train and evaluate!
trainer.train()
metrics = trainer.evaluate()
```

> Remember to push your trained model to the HF hub :) 

Reviewer: ```suggestion
And that's it - you've now trained your first SetFit model! Remember to push your trained model to the Hub :) 
```

```sh
# Push model to the Hub
# Make sure you're logged in with huggingface-cli login first
trainer.push_to_hub("my-awesome-setfit-model")
```

> While this example showed how this can be done with one specific type of base model, any Sentence Transformer base model could be traded in here for different performance and tasks. For intance, using a multilingual Sentence Transformer base can extend few-shot to multilingual settings.

Reviewer: ```suggestion
While this example showed how this can be done with one specific type of base model, any [Sentence Transformer](https://huggingface.co/models?library=sentence-transformers&sort=downloads) model could be traded in here for different performance and tasks. For instance, using a multilingual Sentence Transformer body can extend few-shot classification to multilingual settings.
```

> 

Reviewer: Let's add a small closing :)

```suggestion
## Next steps

So far, we've shown that SetFit is an effective method for few-shot classification tasks. In the coming months, we'll be exploring how well the method generalises to tasks like natural language inference and token classification. In the meantime, we're excited to see how industry practitioners apply SetFit to their use cases - if you have any questions or feedback, open an issue on our [GitHub repo](https://github.com/huggingface/setfit) 🤗!
```

<p>🏎 <strong>Fast to train</strong>: SetFit doesn't require large-scale models like T0 or GPT-3 to achieve high accuracy. As a result, it is typically an order of magnitude (or more) faster to train and run inference with. </p>

<p>🌎 <strong>Multilingual support</strong>: SetFit can be used with any Sentence Transformer on the Hub, which means you can classify text in multiple languages by simply fine-tuning a multilingual checkpoint. </p>

For more details, check out our [paper](https://arxiv.org/abs/2209.11055), [data](https://huggingface.co/SetFit), and [code](https://github.com/huggingface/setfit). In this blog post, we'll explain how SetFit works and how to train your very own models. Let's dive in!

## How does it work?

SetFit is designed with efficiency and simplicity in mind. SetFit first fine-tunes a Sentence Transformer model on a small number of labeled examples (typically 8 or 16 per class). This is followed by training a classifier head on the embeddings generated from the fine-tuned Sentence Transformer. 

<p align="center">
    <img src="assets/103_setfit/setfit_diagram_process.png" width=700>
</p>
<p align="center">
    SetFit's 2-Part Process
</p>

> SetFit takes advantage of sentence transformers’ ability to generate dense embeddings based on paired sentences. In the initial fine-tuning phase stage, it makes use of the limited labeled input data by contrastive training, where positive and negative pairs are created by in-class and out-class selection. The Sentence Transformer model then trains on these pairs (or triplets) and generates dense vectors per example. In the second step, the classification head trains on the encoded embeddings with their respective class labels. At inference time, the unseen example passes through the fine-tuned Sentence Transformer, generating an embedding that when fed to the classification head outputs a class label prediction.

Reviewer: ```suggestion
SetFit takes advantage of Sentence Transformers’ ability to generate dense embeddings based on paired sentences. In the initial fine-tuning phase stage, it makes use of the limited labeled input data by contrastive training, where positive and negative pairs are created by in-class and out-class selection. The Sentence Transformer model then trains on these pairs (or triplets) and generates dense vectors per example. In the second step, the classification head trains on the encoded embeddings with their respective class labels. At inference time, the unseen example passes through the fine-tuned Sentence Transformer, generating an embedding that when fed to the classification head outputs a class label prediction.
```

---
title: "SetFit: Efficient Few-Shot Learning Without Prompts"
thumbnail: /blog/assets/
---
<div class="blog-metadata">
    <small>Published September 16, 2022.</small>
   
<h1>SetFit: Efficient Few-Shot Learning Without Prompts</h1>

</div>

<div class="author-card">
    <a href="/unsojo">
        <div class="bfc">
            <code>unsojo</code>
            <span class="fullname">Unso Jo</span>
        </div>
  </a>
</div>

# SetFit: Efficient Few-Shot Learning Without Prompts
###### The Hugging Face and Intel team introduce a new prompt-free few-shot learning regime called Setfit, with accompanying [code](https://github.com/SetFit/setfit), longer [paper](), and datasets on [HF hub](https://huggingface.co/SetFit)

<kbd>
<img src="assets/gpt3_stone.png">
</kbd>

## Introducing SetFit

> In a collaborative effort between Hugging Face and Intel, we introduce SetFit, a prompt-free few-shot regime made with practicality and efficiency in mind. Recent discussions in ML have focused on few-shot regimes, where only few (zero to a few dozen) data points are needed to extend language model applications to downstream classification tasks. Examples of such regimes include T-few, GPT-3, and ADAPET, but these methods sometimes require large, inaccessible compute and finicky manually crafted prompts. Setfit performs on-par or better than comparable models while being prompt-free and only requiring small models that can fit on commercial personal computers. To promote wide access we make our code and data available open-source. 

Reviewer 2: To avoid repetition, how about alternating between "few-shot regimes" and "few-shot training"?

"inaccessible compute" sounds a little off to me. What about "inaccessible computational resources/infrastructure"?

Setfit --> SetFit

Here's a suggestion for the last two sentences:
SetFit shows similar or improved performance compared to other few-shot methods while also being prompt-free and only requiring small models that can fit on commercial personal computers. We make all code and data used in the development of SetFit publicly available..

<kbd>
<img src="assets/incremental_graph.png">
</kbd>

## SetFit’s Few-Shot Performance

> While run prompt-free and on much smaller base models, SetFit performs on par or better than state of the art few-shot regimes on a variety of benchmarks. On the RAFT, a few-shot benchmark dataset, (https://huggingface.co/spaces/ought/raft-leaderboard)  as of September 2022, SetFit Roberta (with the Roberta-Large ST base model) with 355 million parameters outforms PET and GPT-3 and comes just under average human performance and the 11 billion parameter T-few a model 30 times the size of SetFit Roberta. Footnote (There is no information on state of the art on the RAFT.) SetFit outperforms the human baseline on 7 of the 11 RAFT tasks. 

Reviewer 2: "the RAFT" --> RAFT

SetFit Roberta (using the Roberta-Large ST base model)

"and comes just under average human performance" --> and places just under average human performance

"the 11 billion parameter T-few a model 30" --> the 11 billion parameter T-few, a model 30

| Rank | Method | Accuracy | Model Size | 
| :------: | ------ | :------: | :------: | 
| 1 | YiWise | 76.8 | N/A |
| 2 | T-Few | 75.8 | 11B | 
| 4 | Human Baseline | 73.5 | N/A | 
| 6 | SetFit (Roberta Large) | 71.3 | 355M |
| 9 | PET | 69.6 | 235M |
| 11 | SetFit (MP-Net) | 66.9 | 110M |
| 12 | GPT-3 | 62.7 | 175 B |

> On non-few-shot datasets, SetFit shows robustness across a variety of tasks. It outperforms PERFECT, T-FEW 3 billion, ADAPET and vanilla transformers, on most tasks on sentiment, emotion, counterfactual, and unwanted language classification at very few (n=8) and few (n=64) -shot learning situations. 

Reviewer 2: "language classification at very" --> lanaguage classification tasks

| PERFECT | 34.9 (3.1) | 18.1 (5.3) | 81.5 (8.6) | 26.4  | 75.9 | 64.2 | NA |
|T-Few 3B | NA | 19.0 (3.9) | NA | 57.5 (1.8) | 93.1 (1.6) | NA | NA | 
|SetFit| 43.6(3.0) | 40.3 (11.8) | 88.5 (1.9) | 48.8 (4.5) | 90.1 (3.4) | 82.9 (2.8) | 65.7 (4.6) | 

| Method | SST-5 | AmazonCF | SentEval | Emotion | EnronSpam | AGNews | Average |
| ---- | :----: | :----: | :----: | :----: | :----: | :----: | :----: | 
| Finetune | 45.9 (6.9) | 52.8 (12.1) | 88.9 (1.9) | 65.0 (17.2) | 95.9 (0.8) | 88.4 (0.9) | 72.8 (6.6)|
| ADAPET | 49.9 (1.5) | 56.4 (14.3) | 88.5 (7.4) | 65.0 (17.2) | 93.0 (1.4) | 85.3 (10.4) | 74.9 (6.1)|
| PERFECT | 49.1 (0.7) | 65.1 (5.2) | 91.4  | 66.8  | 92.3 | 88.7 | NA |
|T-Few 3B | NA | 34.7 (4.5) | NA | 71.0 (1.1) | 97.0 (0.3) | NA | NA | 
|SetFit| 51.9 (0.6) | 61.9 (2.9) | 90.4 (0.6) | 76.2 (1.3) | 96.1 (0.8) | 88.0 (0.7) | 77.4 (1.2) | 

And just by switching out the base ST model to a multilingual one, SetFit can function seamlessly in multilingual contexts. In our experiments, SetFit’s performance shows promising results on classification in German, Japanese, Mandarin, French and Spanish, in both in-language and cross linguistic settings.

## What is SetFit?

> The strength of SetFit is its efficiency and simplicity. SetFit first finetunes a Sentence Transformer (ST) model then trains a classifier head on the generated ST embeddings. 

Reviewer 2: "trains a classifier head on the generated ST embeddings." --> trains a classifier head on the embeddings generated from the finetuned ST.

<h1>SetFit: Efficient Few-Shot Learning Without Prompts</h1>

</div>
<div class="author-card">
    <a href="/eunseojo">

>         <img class="avatar avatar-user" src="https://scholar.googleusercontent.com/citations?view_op=medium_photo&user=-I2EZeEAAAAJ&citpid=8" title="Gravatar">

Reviewer: Should we add profile pics of everyone in the colab?
